Source/bindings/modules/v8/custom/V8WebCLCommandQueueCustom.cpp

```diff
@@ -0,0 +1,1207 @@
+        exceptionState.throwTypeError(ExceptionMessages::notEnoughArguments(3, info.Length()));
+        exceptionState.throwIfNeeded();
+        return;
+    }
+    exceptionState.throwWebCLException(WebCLException::INVALID_ARG_VALUE, WebCLException::invalidArgValueMessage);
+    exceptionState.throwIfNeeded();
+}
+
+void V8WebCLCommandQueue::enqueueNDRangeKernelMethodCustom(const v8::FunctionCallbackInfo<v8::Value>& info)
+{
+    ExceptionState exceptionState(ExceptionState::ExecutionContext, "enqueueNDRangeKernel", "WebCLCommandQueue", info.Holder(), info.GetIsolate());
+    if (UNLIKELY(info.Length() < 4)) {
+        setMinimumArityTypeError(exceptionState, 4, info.Length());
+        exceptionState.throwIfNeeded();
+        return;
+    }
+    WebCLCommandQueue* impl = V8WebCLCommandQueue::toImpl(info.Holder());
+    WebCLKernel* kernel;
+    unsigned workDim;
+    Vector<double> offsets;
+    Vector<double> globalWorkSize;
+    Vector<double> localWorkSize;
+    Vector<RefPtr<WebCLEvent>> eventWaitList;
+    WebCLEvent* event = nullptr;
+    {
+        if (info.Length() > 0 && !V8WebCLKernel::hasInstance(info[0], info.GetIsolate())) {
+            exceptionState.throwTypeError("parameter 1 is not of type 'WebCLKernel'.");
+            exceptionState.throwIfNeeded();
+            return;
+        }
+        kernel = V8WebCLKernel::toImplWithTypeCheck(info.GetIsolate(), info[0]);
+        TONATIVE_VOID_EXCEPTIONSTATE_INTERNAL(workDim, toUInt32(info[1], EnforceRange, exceptionState), exceptionState);
+        if (info.Length() > 2 && !isUndefinedOrNull(info[2]))
+            TONATIVE_VOID_EXCEPTIONSTATE_INTERNAL(offsets, toImplArray<double>(info[2], 3, info.GetIsolate(), exceptionState), exceptionState);
+        TONATIVE_VOID_EXCEPTIONSTATE_INTERNAL(globalWorkSize, toImplArray<double>(info[3], 4, info.GetIsolate(), exceptionState), exceptionState);
+        if (info.Length() > 4 && !isUndefinedOrNull(info[4]))
+            TONATIVE_VOID_EXCEPTIONSTATE_INTERNAL(localWorkSize, toImplArray<double>(info[4], 5, info.GetIsolate(), exceptionState), exceptionState);
+        if (info.Length() > 5 && !isUndefinedOrNull(info[5]))
+            TONATIVE_VOID_EXCEPTIONSTATE_INTERNAL(eventWaitList, (toRefPtrNativeArray<WebCLEvent, V8WebCLEvent>(info[5], 6, info.GetIsolate(), exceptionState)), exceptionState);
+        if (info.Length() > 6) {
+            if (!isUndefinedOrNull(info[6]) && !V8WebCLEvent::hasInstance(info[6], info.GetIsolate())) {
+                exceptionState.throwTypeError("parameter 7 is not of type 'WebCLEvent'.");
+                exceptionState.throwIfNeeded();
+                return;
+            }
+            event = V8WebCLEvent::toImplWithTypeCheck(info.GetIsolate(), info[6]);
+        }
+    }
+    impl->enqueueNDRangeKernel(kernel, workDim, offsets, globalWorkSize, localWorkSize, eventWaitList, event, exceptionState);
+    if (exceptionState.hadException()) {
+        exceptionState.throwIfNeeded();
+        return;
```

Comment:
Nit, You have lots return at last time which are unnecessary. Could you go through code and remove those?